aaelf64-morello/aaelf64-morello.rst

```diff
@@ -759,6 +773,19 @@ Dynamic Morello relocations
 
   ``| 64-bits offset | 64-bits size |``
 
+  ``R_MORELLO_CODE_CAPINIT`` is similar to ``R_MORELLO_CAPINIT`` but signifies
+  that the capability being relocated is a code pointer as opposed to a data
+  pointer or function pointer. This distinction is needed for library-based
+  compartmentalization (c18n). Note that current toolchain only allows code
+  pointers to be taken for non-preemptible symbols, so all instances of
+  ``R_MORELLO_CODE_CAPINIT`` would be converted to relative relocations at
+  static-link time. This behaviour is not guaranteed in the future.
```

Comment:
```suggestion
  compartmentalization (c18n). Currently for security reasons this relocation may only be used for non-preemptible symbols, and thus will be converted to ``R_MORELLO_RELATIVE`` at link time, making it in effect a static relocation. However, this restriction may be removed in future if needed, and is therefore classed as a dynamic relocation.
```

Also, is it an error to use this relocation against a non-STT_FUNC symbol? If so we should state that. (Think carefully about STT_GNU_IFUNC, not just data/untyped symbols)

Reply:
It is not an error to use this for a non-STT_FUNC symbol, and the toolchain will just convert it to a relative relocation (or error out for global symbols).

Reply:
I'll rephrase: is there a legitimate use case for permitting such uses, or should they be forbidden by the spec?

Reply:
We should probably forbid it - I cannot think of any legitimate use cases.